src/domain/core/src/use_cases/create_dataset_from_snapshot_use_case.rs

```diff
@@ -18,7 +18,48 @@ pub trait CreateDatasetFromSnapshotUseCase: Send + Sync {
     async fn execute(
         &self,
         snapshot: DatasetSnapshot,
+        options: &CreateDatasetFromSnapshotUseCaseOptions,
```

Comment:
What is the point of passing by ref? Most of the time you are using temporary objects, like `Default::default()`

Reply:
Indeed -- especially since these objects may not even be merely clonable, but copyable

Fixed